Add warning installation of GatewayAPI CRDs (#33974)
cilium-team/cilium: Documentation/network/servicemesh/gateway-api/installation.rst

    - `ReferenceGrant <https://gateway-api.sigs.k8s.io/api-types/referencegrant/>`_
    - `TLSRoute (experimental) <https://gateway-api.sigs.k8s.io/references/spec/#gateway.networking.k8s.io%2fv1alpha2.TLSRoute/>`_

>     .. code-block:: shell-session

Reviewer: Why change the indent level and move this code block out of the list item?

Author: good catch! will fix it

> .. admonition:: Warning

Reviewer: Similarly, let's keep the warning inside of the list?
And let's use a lowercase `w` for consistency with other admonitions.

```suggestion
    .. admonition:: warning
```

    :class: attention

>     Currently, an issue in Gateway API requires Cilium having the experimental CRDs to be installed in order to use Gateway API. The issue is being tracked `here <https://github.com/kubernetes-sigs/gateway-api/issues/3080>`__.

Reviewer: “Currently” is only guaranteed to be accurate at the time of writing. Let's remove it.

```suggestion
    An issue in Gateway API requires Cilium having the experimental CRDs to be installed in order to use Gateway API. The issue is being tracked `here <https://github.com/kubernetes-sigs/gateway-api/issues/3080>`__.
```

>     To install the experimental CRDs, please refer to this `doc <https://gateway-api.sigs.k8s.io/guides/#install-experimental-channel>`_ 

Reviewer: Let's be more precise.

```suggestion
    To install the experimental CRDs, please refer to the `Gateway API documentation <https://gateway-api.sigs.k8s.io/guides/#install-experimental-channel>`__.
```
